src/dotnet/commands/dotnet-cache/CacheCommand.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Tools.MSBuild;
+using Microsoft.DotNet.Tools.Restore;
+using Microsoft.DotNet.Cli.CommandLine;
+
+namespace Microsoft.DotNet.Tools.Cache
+{
+    public partial class CacheCommand
+    {
+        public string ProjectPath { get; set; }
+        public string Framework { get; set; }
+        public string Runtime { get; set; }
+        public string OutputPath { get; set; }
+        public string FxVersion { get; set; }
+        public string WorkingDir { get; set; }
+        public string Verbosity { get; set; }
+        private bool PreserveWorkingDir { get; set; }
+
+        
+        public List<string> ExtraMSBuildArguments { get; set; }
+
+        private CacheCommand()
+        {
+        }
+
+        private void PrepareForExecute(CommandArgument argument)
+        {
+            string projectPath = string.Empty;
+            try
+            {
+
+                if (string.IsNullOrEmpty(WorkingDir))
+                {
+                    var tempDir = Path.GetTempPath();
+                    var folder  = Path.GetRandomFileName();
+                    WorkingDir  = Path.Combine(tempDir, folder);
+                }
+                else
+                {
+                    WorkingDir = Path.GetFullPath(WorkingDir);
+                }
+
+                if (Directory.Exists(WorkingDir))
+                {
+                    Directory.Delete(WorkingDir, true);
+                }
+
+                Directory.CreateDirectory(WorkingDir);
+                var givenProject = argument.Value;
+                var projectFile = Path.GetFileName(givenProject);
+                projectPath = Path.Combine(WorkingDir, projectFile);
+
+                File.Copy(givenProject, projectPath);
+
+                ProjectPath = projectPath;
+            }
+            catch (System.Exception e)
+            {
+                throw new GracefulException(e.ToString());
+            }
+        }
+        private void CleanupAfterExecute()
+        {
+            if (!PreserveWorkingDir)
+            {
+                try
+                {
+                    Directory.Delete(WorkingDir, true);
+                }
+                catch (System.Exception e)
+                {
+                    throw new GracefulException(e.ToString());
+                }
+            }
+        }
+
+        public int Execute()
+        {
+            int result = 0;
+            List<string> msbuildArgs = new List<string> ();
+            if (!string.IsNullOrEmpty(ProjectPath))
+            {
+                msbuildArgs.Add(ProjectPath);
+            }
+
+            msbuildArgs.Add("/t:ComposeCache");
+
+            if (!string.IsNullOrEmpty(Framework))
```

Comment:
"TFM" shouldn't be a mandatory argument.  When there is only one, it should use what is already listed in the project.  The user on the command line shouldn't have to pass in what is already in the .csproj.

Same for "OutputPath" below - by default use what is in the project.